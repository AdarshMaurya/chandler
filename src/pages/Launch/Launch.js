import React from 'react';
import PropTypes from 'prop-types';
import {Text, Image, View} from 'react-native';
import {SafeAreaView} from 'react-navigation';

import {commonStyles, forceInset, preAuthScreensStyles} from 'theme';
import {Button, BottomNav} from 'components';

export const Launch = ({navigate}) => (
  <SafeAreaView forceInset={forceInset} style={preAuthScreensStyles.container}>
    <View style={commonStyles.flex} />

    <View style={preAuthScreensStyles.centeredBlock}>
      <Image source={require('./assets/launch.png')} />

      <Text style={preAuthScreensStyles.primaryText}>
        Keep track of every interaction with your friend and family!
      </Text>
      <Text style={preAuthScreensStyles.secondaryText}>
        Remember when you last called, what you talked about, the name of their
        kids or what they prefer to eat!
      </Text>

      <Button onPress={navigate('Signup')}>Get started</Button>
    </View>

    <View style={commonStyles.flex} />

    <BottomNav
      title="Already have an account ?"
      linkTitle="Sign in"
      onPress={navigate('Login')}
    />
  </SafeAreaView>
);

Launch.propTypes = {
  navigate: PropTypes.func.isRequired,
};
