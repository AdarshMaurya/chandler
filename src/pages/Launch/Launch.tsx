import React, {PureComponent} from 'react';
import {Image, Text, View} from 'react-native';
import SafeAreaView from 'react-native-safe-area-view';

import {Button} from '@components';
import {commonStyles, forceInset, preAuthScreensStyles} from '@theme';

interface ILaunchProps {
  navigate: (routeName) => any;
}

export class Launch extends PureComponent<ILaunchProps, {}> {
  public render() {
    const {navigate} = this.props;

    return (
      <SafeAreaView
        forceInset={forceInset}
        style={preAuthScreensStyles.container}
      >
        <View style={commonStyles.flex} />

        <View style={preAuthScreensStyles.centeredBlock}>
          <Image source={require('./assets/launch.png')} />

          <Text style={preAuthScreensStyles.primaryText}>
            Keep track of every interaction with your friend and family!
          </Text>
          <Text style={preAuthScreensStyles.secondaryText}>
            Remember when you last called, what you talked about, the name of
            their kids or what they prefer to eat!
          </Text>

          <Button onPress={navigate('Login')}>Get started</Button>
        </View>

        <View style={commonStyles.flex} />

        {/* <BottomNav
      title="Already have an account ?"
      linkTitle="Sign in"
      onPress={navigate('Login')}
    /> */}
      </SafeAreaView>
    );
  }
}
