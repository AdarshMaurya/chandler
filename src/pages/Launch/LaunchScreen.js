import {connect} from 'react-redux';

import {navigate} from 'redux/router';
import {Launch} from './Launch';

export const LaunchScreen = connect(null, dispatch => {
  return {
    navigateLogin: () => dispatch(navigate('Login')),
    navigateSignup: () => dispatch(navigate('Signup')),
  };
})(Launch);
