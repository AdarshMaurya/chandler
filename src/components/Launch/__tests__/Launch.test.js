import React from 'react';
import {shallow} from 'enzyme';
import {Launch} from '../Launch';

describe('Components', () => {
  describe('Launch', () => {
    it('should renders correctly', () => {
      const tree = shallow(<Launch />);
      expect(tree).toMatchSnapshot();
    });
  });
});
