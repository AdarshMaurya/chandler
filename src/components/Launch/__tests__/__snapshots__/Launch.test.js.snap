// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components Launch should renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Launch />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        />,
        <View
          style={
            Object {
              "paddingHorizontal": 30,
              "width": "100%",
            }
          }
        >
          <Image
            source={1}
          />
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 20,
                "marginTop": 40,
                "textAlign": "center",
              }
            }
          >
            Keep track of every interaction with your friend and family!
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#97A5B2",
                "marginBottom": 20,
                "textAlign": "center",
              }
            }
          >
            Remember when you last called, what you talked about, the name of their kids or what they prefer to eat!
          </Text>
          <Button
            onPress={[Function]}
          >
            Get started
          </Button>
        </View>,
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        />,
        <View
          style={
            Object {
              "borderColor": "#E2E4E5",
              "borderTopWidth": 0.5,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingVertical": 15,
              "width": "100%",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            Already have an account ? 
          </Text>
          <TouchableOpacity
            activeOpacity={0.2}
            hitSlop={
              Object {
                "bottom": 15,
                "left": 15,
                "right": 15,
                "top": 15,
              }
            }
            onPress={[Function]}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#1361cb",
                }
              }
            >
              Sign in
            </Text>
          </TouchableOpacity>
        </View>,
      ],
      "forceInset": Object {
        "top": "always",
      },
      "style": Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Image
              source={1}
            />,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 20,
                  "marginTop": 40,
                  "textAlign": "center",
                }
              }
            >
              Keep track of every interaction with your friend and family!
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#97A5B2",
                  "marginBottom": 20,
                  "textAlign": "center",
                }
              }
            >
              Remember when you last called, what you talked about, the name of their kids or what they prefer to eat!
            </Text>,
            <Button
              onPress={[Function]}
            >
              Get started
            </Button>,
          ],
          "style": Object {
            "paddingHorizontal": 30,
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "source": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Keep track of every interaction with your friend and family!",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 18,
                "fontWeight": "bold",
                "marginBottom": 20,
                "marginTop": 40,
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": "Keep track of every interaction with your friend and family!",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Remember when you last called, what you talked about, the name of their kids or what they prefer to eat!",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#97A5B2",
                "marginBottom": 20,
                "textAlign": "center",
              },
            },
            "ref": null,
            "rendered": "Remember when you last called, what you talked about, the name of their kids or what they prefer to eat!",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Get started",
              "onPress": [Function],
            },
            "ref": null,
            "rendered": "Get started",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "style": Object {
            "flex": 1,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Already have an account ? 
            </Text>,
            <TouchableOpacity
              activeOpacity={0.2}
              hitSlop={
                Object {
                  "bottom": 15,
                  "left": 15,
                  "right": 15,
                  "top": 15,
                }
              }
              onPress={[Function]}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#1361cb",
                  }
                }
              >
                Sign in
              </Text>
            </TouchableOpacity>,
          ],
          "style": Object {
            "borderColor": "#E2E4E5",
            "borderTopWidth": 0.5,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingVertical": 15,
            "width": "100%",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Already have an account ? ",
              "ellipsizeMode": "tail",
            },
            "ref": null,
            "rendered": "Already have an account ? ",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#1361cb",
                  }
                }
              >
                Sign in
              </Text>,
              "hitSlop": Object {
                "bottom": 15,
                "left": 15,
                "right": 15,
                "top": 15,
              },
              "onPress": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Sign in",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#1361cb",
                },
              },
              "ref": null,
              "rendered": "Sign in",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          />,
          <View
            style={
              Object {
                "paddingHorizontal": 30,
                "width": "100%",
              }
            }
          >
            <Image
              source={1}
            />
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 20,
                  "marginTop": 40,
                  "textAlign": "center",
                }
              }
            >
              Keep track of every interaction with your friend and family!
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#97A5B2",
                  "marginBottom": 20,
                  "textAlign": "center",
                }
              }
            >
              Remember when you last called, what you talked about, the name of their kids or what they prefer to eat!
            </Text>
            <Button
              onPress={[Function]}
            >
              Get started
            </Button>
          </View>,
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          />,
          <View
            style={
              Object {
                "borderColor": "#E2E4E5",
                "borderTopWidth": 0.5,
                "flexDirection": "row",
                "justifyContent": "center",
                "paddingVertical": 15,
                "width": "100%",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
            >
              Already have an account ? 
            </Text>
            <TouchableOpacity
              activeOpacity={0.2}
              hitSlop={
                Object {
                  "bottom": 15,
                  "left": 15,
                  "right": 15,
                  "top": 15,
                }
              }
              onPress={[Function]}
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#1361cb",
                  }
                }
              >
                Sign in
              </Text>
            </TouchableOpacity>
          </View>,
        ],
        "forceInset": Object {
          "top": "always",
        },
        "style": Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Image
                source={1}
              />,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontSize": 18,
                    "fontWeight": "bold",
                    "marginBottom": 20,
                    "marginTop": 40,
                    "textAlign": "center",
                  }
                }
              >
                Keep track of every interaction with your friend and family!
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#97A5B2",
                    "marginBottom": 20,
                    "textAlign": "center",
                  }
                }
              >
                Remember when you last called, what you talked about, the name of their kids or what they prefer to eat!
              </Text>,
              <Button
                onPress={[Function]}
              >
                Get started
              </Button>,
            ],
            "style": Object {
              "paddingHorizontal": 30,
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "source": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Keep track of every interaction with your friend and family!",
                "ellipsizeMode": "tail",
                "style": Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "marginBottom": 20,
                  "marginTop": 40,
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": "Keep track of every interaction with your friend and family!",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Remember when you last called, what you talked about, the name of their kids or what they prefer to eat!",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#97A5B2",
                  "marginBottom": 20,
                  "textAlign": "center",
                },
              },
              "ref": null,
              "rendered": "Remember when you last called, what you talked about, the name of their kids or what they prefer to eat!",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Get started",
                "onPress": [Function],
              },
              "ref": null,
              "rendered": "Get started",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "style": Object {
              "flex": 1,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              >
                Already have an account ? 
              </Text>,
              <TouchableOpacity
                activeOpacity={0.2}
                hitSlop={
                  Object {
                    "bottom": 15,
                    "left": 15,
                    "right": 15,
                    "top": 15,
                  }
                }
                onPress={[Function]}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#1361cb",
                    }
                  }
                >
                  Sign in
                </Text>
              </TouchableOpacity>,
            ],
            "style": Object {
              "borderColor": "#E2E4E5",
              "borderTopWidth": 0.5,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingVertical": 15,
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Already have an account ? ",
                "ellipsizeMode": "tail",
              },
              "ref": null,
              "rendered": "Already have an account ? ",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 0.2,
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#1361cb",
                    }
                  }
                >
                  Sign in
                </Text>,
                "hitSlop": Object {
                  "bottom": 15,
                  "left": 15,
                  "right": 15,
                  "top": 15,
                },
                "onPress": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Sign in",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": "#1361cb",
                  },
                },
                "ref": null,
                "rendered": "Sign in",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
